_overviews/overview/control-structures.md

```diff
@@ -0,0 +1,555 @@
+
+You can also use `for` loops with a `Map`. For example, given this `Map` of movie names and ratings:
+
+<!-- TODO: use a different example -->
+```scala
+val ratings = Map(
+  "Lady in the Water"  -> 3.0, 
+  "Snakes on a Plane"  -> 4.0, 
+  "You, Me and Dupree" -> 3.5
+)
+```
+
+You can print the movie names and ratings using `for` like this:
+
+```scala
+for (name,rating) <- ratings do println(s"Movie: $name, Rating: $rating")
+```
+
+Here’s what that looks like in the REPL:
+
+```scala
+scala> for (name,rating) <- ratings do println(s"Movie: $name, Rating: $rating")
+Movie: Lady in the Water, Rating: 3.0
+Movie: Snakes on a Plane, Rating: 4.0
+Movie: You, Me and Dupree, Rating: 3.5
+```
+
+As the `for` loop iterates over the map, each rating is bound to the variables `name` and `rating`, which are in a tuple:
+
+```scala
+(name,rating) <- ratings
+```
+
+As the loop runs, the variable `name` is assigned to the current _key_ in the map, and the variable `rating` is assigned to the current map _value_.
+
+<!-- TODO: show other possible uses/styles here?
+for (name,rating) <- ratings
+do println(s"Movie: $name, Rating: $rating")
+
+for
+  (name,rating) <- ratings
+do
+  println(s"Movie: $name, Rating: $rating")
+
+for (name,rating) <- ratings do { println(s"Movie: $name, Rating: $rating") }
+for ((name,rating) <- ratings) println(s"Movie: $name, Rating: $rating")
+-->
+
+
+
+## `for` expressions
+
+In the previous `for` loop examples, those loops were all used for _side effects_, specifically to print those values to STDOUT using `println`.
+
+It’s important to know that you can also create `for` _expressions_ that return values. You create a `for` expression by adding the `yield` keyword and an expression to return, like this:
+
+```scala
+val list = 
+  for
+    i <- 10 to 12
+  yield
+    i * 2
+
+// result: list == Vector(20,22,24)
+```
+
+After that `for` expression runs, the variable `list` is a `Vector` that contains the values shown. This is how the expression works:
+
+1. The `for` expression starts to iterate over the values in the range `(10, 11, 12)`. It first works on the value `10`, multiplies it by `2`, then _yields_ that result, the value `20`.
+2. Next, it works on the `11` — the second value in the range. It multiples it by `2`, then yields the value `22`. You can think of these yielded values as accumulating in a temporary holding place.
+3. Finally the loop gets the number `12` from the range, multiplies it by `2`, yielding the number `24`. The loop completes at this point and yields the final result, the `Vector(20,22,24)`.
+
+`for` expressions can be used any time you need to traverse all of the elements in a collection and apply an algorithm to those elements to create a new list.
+
+Here’s an example that shows how to use a block of code after the `yield`:
+
+```scala
+val names = List("_adam", "_david", "_frank")
+
+val capNames = for name <- names yield
+  // imagine this algorithm requires multiple lines
```

Comment:
I may have updated this at some point, but the current code can be reduced from this:

```scala
val capNames = for name <- names yield
  // imagine this algorithm requires multiple lines
  val nameWithoutUnderscore = name.drop(1)
  val capName = nameWithoutUnderscore.capitalize
  capName
```

to this:

```scala
val capNames = for name <- names yield name.drop(1).capitalize
```

So that’s why I have that comment in there. I’ll try to come up with a better example, or word that better.